storage/store/inmemory/unsynchronized/registers.go

```diff
@@ -0,0 +1,67 @@
+package unsynchronized
+
+import (
+	"go.uber.org/atomic"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+)
+
+type RegisterEntries map[flow.RegisterID]flow.RegisterValue
+type HeightToRegisterEntries map[uint64]RegisterEntries
+
+type Registers struct {
+	firstHeight  uint64
+	latestHeight atomic.Uint64
+	store        HeightToRegisterEntries
+}
+
+func NewRegisters(firstHeight uint64) *Registers {
+	return &Registers{
+		firstHeight:  firstHeight,
+		latestHeight: atomic.Uint64{},
+		store:        make(HeightToRegisterEntries),
+	}
+}
+
+var _ storage.RegisterIndex = (*Registers)(nil)
+
+func (r *Registers) Get(ID flow.RegisterID, height uint64) (flow.RegisterValue, error) {
+	h, ok := r.store[height]
+	if !ok {
+		return flow.RegisterValue{}, storage.ErrNotFound
+	}
+
+	val, ok := h[ID]
+	if !ok {
+		return flow.RegisterValue{}, storage.ErrNotFound
+	}
+
+	return val, nil
+}
+
+func (r *Registers) LatestHeight() uint64 {
+	return r.latestHeight.Load()
+}
+
+func (r *Registers) FirstHeight() uint64 {
+	return r.firstHeight
+}
+
+func (r *Registers) Store(entries flow.RegisterEntries, height uint64) error {
+	if height > r.latestHeight.Load() {
```

Comment:
This should replicate the checks in the original store and return an error if `height != r.latestHeight+1`, and return nil if `height == r.latestHeight`